### ISSUE TITLE
It would have been better to make `progress_bar` a keyword only argument everywhere

### ISSUE THREAD

**Issue by tkoyama010** [pyvista/pyvista#5506]
### Describe what maintenance you would like added.

If considering adding new arguments, it would have been better to make `progress_bar` a keyword only argument everywhere.

_Originally posted by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5502#discussion_r1462236499_

### Links to source code.

None

### Pseudocode or Screenshots

None

**Comment by user27182** [comment]
I agree with enforcing keyword only args, especially for something like `progress_bar`.

I think this applies more broadly to a lot of the code in PyVista. There are many keyword args which are likely never used in practice as positional args, but are technically positional (and therefore would introduce breaking changes if new kwargs are introduced anywhere but at the end of the kwargs dict). Many of these probably should be enforced as keyword only anyway since it can be easy to specify args for a different keyword than intended (and introduce a bug as a result).

For example, all the kwargs in `plot` are positional. This means it's possible to do something like:
``` python
pv.PolyData([0.0,0.0,0.0]).plot(False, False, False, False)
```
But using positional args here is not very helpful here since there are many boolean parameters for `plot`. In addition to eliminating backwards-compatibility issues, enforcing keyword-only args would result in much more explicit code:

``` python
pv.PolyData([0.0,0.0,0.0]).plot(off_screen=False, full_screen=False, screenshot=False, interactive=False)
```

**Comment by darikg** [comment]
I think a number of scientific python libraries went mostly keyword only, [sklearn](https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep009/proposal.html) for example. They have a [decorator](https://github.com/thomasjpfan/scikit-learn/blob/a48f0eafd13ec6cbc4ab6a8b08c92588f9cc8973/sklearn/utils/validation.py#L39) to issue positional deprecation warnings/errors.

**Comment by MatthewFlamm** [comment]
The major point that gives me pause for enforcing this for *all* function args, or a majority, is the breaking change required.  If it can be done with warnings and a deprecation like the example given, then I'm +1 for implementing it.  If we do it, then we should align on what should be positional vs. keyword only and do it within one release cycle.
